From: Brian Vazquez <brianvv@google.com>
Date: Wed, 15 Jan 2020 10:43:03 -0800
Subject: bpf: Add lookup and update batch ops to arraymap
Patch-mainline: v5.6-rc1
Git-commit: c60f2d2861778de6370a4f4ca6ab1d7d4a32efae
References: bsc#1177028

This adds the generic batch ops functionality to bpf arraymap, note that
since deletion is not a valid operation for arraymap, only batch and
lookup are added.

Signed-off-by: Brian Vazquez <brianvv@google.com>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Yonghong Song <yhs@fb.com>
Link: https://lore.kernel.org/bpf/20200115184308.162644-5-brianvv@google.com
Acked-by: Gary Lin <glin@suse.com>
---
 kernel/bpf/arraymap.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/kernel/bpf/arraymap.c
+++ b/kernel/bpf/arraymap.c
@@ -503,6 +503,8 @@ const struct bpf_map_ops array_map_ops =
 	.map_mmap = array_map_mmap,
 	.map_seq_show_elem = array_map_seq_show_elem,
 	.map_check_btf = array_map_check_btf,
+	.map_lookup_batch = generic_map_lookup_batch,
+	.map_update_batch = generic_map_update_batch,
 };
 
 const struct bpf_map_ops percpu_array_map_ops = {
